Fix index.php links in admin zone

Hlavtox opened the pull request:
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Simmilar to https://github.com/PrestaShop/PrestaShop/pull/35167, I migrated all usages of hardcoded index.php in admin zone to getAdminLink method. I needed to fixed some things along the way. There are million places with hardcoded parameters behind the calls still, but that would be next step sometime.
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See below
| UI Tests          | https://github.com/Hlavtox/ga.tests.ui.pr/actions/runs/7812733796
| Fixed issue or discussion?     | 
| Related PRs       | 
| Sponsor company   | 

### How to test
- Check you can upload carrier image in BO.
- Check that links to customer and orders in customer threads page in BO work correctly.
- Check that if you go configure any module, the Translate button in the header works properly.
- Check that search in the BO on the top works correctly.

Hlavtox committed:
Fix index.php links in admin zone

Hlavtox commented:
Test run - https://github.com/Hlavtox/ga.tests.ui.pr/actions/runs/7738648814

matks reviewed (commented):
@Hlavtox I think "Fix index.php links in admin zone" is not an accurate description of this PR content 😉

matks reviewed (dismissed):
@Hlavtox I think this PR needs to be tested by a QA analyst. And there is too much different things in this PR 🤔 there's refactoring, link changes, variables removal... I think it will be hard for a QA analyst to verify all of this.

I suggest to split the PR in smaller parts, each of them with a clear focus / goal and a clear "How to test". You are modifying a lot of legacy code, that is fragile and prone to errors. Please rely on QA team skills to verify whether or not you have broken something.

Hlavtox commented:
@matks I don't know what I should do now, why did you put a block? 😁 I can just edit the PR table and let QA test it ;)

matks commented:
@Hlavtox I put a stop for the "I tested all sketchy things manually, green tests should be sufficient. :-)" in "How to test" 😉 

Yes please modify it and request the help of QA team to make sure nothing is broken.

But I'm still worried: there's refactoring, link changes, variables removal... I think it will be hard for a QA analyst to verify all of this. A lot of things are mixed together.

Hlavtox commented:
@matks I changed the PR table to account for it.

Regarding the changes, it's very simple:
- `admin-dev/init.php` is used only by the filemanager. In that file, there was a hardcoded link. When I changed it to proper URL generated, it didn't work, because `admin-dev/filemanager/config/config.php` had it's own assignment of _PS_ADMIN_DIR_, which was different from the admin one. Because of that, the method was generating links like `https://localhost/dev/../index.php` or something like that.
- `classes/helper/HelperList.php` - currentIndex can never be AdminProducts since it's now removed
- `trad_link` is not needed, because we can determine if there are any translation links by checking for emptiness of `$translateLinks` array.

florine2623 reviewed (approved):
Hello @Hlavtox ,

Tested the following :

- Check you can upload carrier image in BO ✅ tested in Creation and Edition of carrier, in FO carrier is well displayed
- Check that links to customer and orders in customer threads page in BO work correctly ✅ check links in Threads and in Orders and messages timeline
- Check that if you go configure any module, the Translate button in the header works properly ✅ Translate button redirects to the right translation page, translation is well saved
- Check that search in the BO on the top works correctly ✅ modules, customer, orders, feature.

LGTM

Hlavtox commented:
Thank you @florine2623! :-)
